### PR TITLE
Add sign out screen

### PR DESCRIPTION
**Pull request by flavius-mester**
Sign out:
 - added SignOut use cases that removes all of the required data
 - added a separate SignOut Route (mostly as its unrelated from the rest of the routes, and it needs to be global - it can be triggered whenever)
 
 

Screenshot_20240628_093442_OneLogin

### CODE REVIEW

File: app/src/test/java/uk/gov/onelogin/tokens/usecases/RemoveAllSecureStoreDataTest.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        verify(mockSecureStore).delete(
+            Keys.ACCESS_TOKEN_KEY,
+            mockFragmentActivity
+        )
+        verify(mockSecureStore).delete(
+            Keys.ID_TOKEN_KEY,
+            mockFragmentActivity
+        )
+    }
```
Comment (alex-bradbury) [inline app/src/test/java/uk/gov/onelogin/tokens/usecases/RemoveAllSecureStoreDataTest.kt:39]: Add test here for the caught exception...?

File: app/src/androidTest/java/uk/gov/onelogin/signOut/ui/SignOutScreenKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    fun verifyScreenDisplayed() {
+        composeTestRule.onNode(title).assertIsDisplayed()
+    }
+
+    @Test
+    fun verifySignOutButton() {
+        composeTestRule.onNode(button).performClick()
+        verify(signOutUseCase).invoke(composeTestRule.activity)
+    }
```
Comment (alex-bradbury) [inline app/src/androidTest/java/uk/gov/onelogin/signOut/ui/SignOutScreenKtTest.kt:47]: Can we add test for the `goBack` on click?

File: app/src/test/java/uk/gov/onelogin/tokens/usecases/RemoveAllSecureStoreDataTest.kt (hunk 1 above)
Comment (flavius-mester) [reply app/src/test/java/uk/gov/onelogin/tokens/usecases/RemoveAllSecureStoreDataTest.kt:42]: done 03d1a15dbfc214ab24561c222d60e59fc057ed22

File: app/src/androidTest/java/uk/gov/onelogin/signOut/ui/SignOutScreenKtTest.kt (hunk 2 above)
Comment (flavius-mester) [reply app/src/androidTest/java/uk/gov/onelogin/signOut/ui/SignOutScreenKtTest.kt:58]: done 03d1a15dbfc214ab24561c222d60e59fc057ed22

File: app/src/main/java/uk/gov/onelogin/ui/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -42,7 +42,10 @@ import uk.gov.onelogin.ui.components.EmailHeader
 import uk.gov.onelogin.ui.components.LightRed
 
 @Composable
-fun ProfileScreen(viewModel: ProfileScreenViewModel = hiltViewModel()) {
+fun ProfileScreen(
+    openSignOutScreen: () -> Unit = { },
```
[the remark this answers is not in the document]
Comment (flavius-mester) [reply app/src/main/java/uk/gov/onelogin/ui/profile/ProfileScreen.kt:46]: done 03d1a15dbfc214ab24561c222d60e59fc057ed22

**Pull request merged**